controller/namedspaces_test.go

```diff
@@ -146,3 +149,115 @@ func (rest *TestNamedSpaceREST) TestShow() {
 		assert.Equal(t, &fxt.Spaces[0].Description, namedspace.Data.Attributes.Description)
 	})
 }
+
+func (rest *TestNamedSpaceREST) TestSuccessSortListSpaces() {
+	t := rest.T()
+	resource.Require(t, resource.Database)
+
+	// names of the spaces to be created
+	inputSpaces := []string{
+		"Test 24",
+		"Test 22",
+		"Test 23",
+	}
+	// randominze the space names so that they don't clash
+	random := testsupport.CreateRandomValidTestName("")
+	for i := range inputSpaces {
+		inputSpaces[i] += random
+	}
+
+	// create actual spaces with those names
+	fxt := tf.NewTestFixture(t, rest.DB, tf.Spaces(len(inputSpaces),
+		func(fxt *tf.TestFixture, idx int) error {
+			fxt.Spaces[idx].Name = inputSpaces[idx]
+			return nil
+		}),
+	)
+	identity := *fxt.Identities[0]
+
+	// Now try to list the data in various circumstances
+	tests := []struct {
+		name      string
+		sortParam *string
+		want      []string
+	}{
+		{
+			name: "no sort param given",
+			want: []string{
+				"Test 23",
+				"Test 22",
+				"Test 24",
+			},
+		},
+		{
+			name:      "sort in ascending using name",
+			sortParam: ptr.String("name"),
+			want: []string{
+				"Test 22",
+				"Test 23",
+				"Test 24",
+			},
+		},
+		{
+			name:      "sort in descending using name",
+			sortParam: ptr.String("-name"),
+			want: []string{
+				"Test 24",
+				"Test 23",
+				"Test 22",
+			},
+		},
+		{
+			name:      "sort in ascending using created at",
+			sortParam: ptr.String("created"),
+			want: []string{
+				"Test 24",
+				"Test 22",
+				"Test 23",
+			},
+		},
+		{
+			name:      "sort in descending using created at",
+			sortParam: ptr.String("-created"),
+			want: []string{
+				"Test 23",
+				"Test 22",
+				"Test 24",
+			},
+		},
+		{
+			name:      "sort in ascending using updated at",
+			sortParam: ptr.String("updated"),
+			want: []string{
+				"Test 24",
+				"Test 22",
+				"Test 23",
+			},
+		},
+		{
+			name:      "sort in descending using updated at",
+			sortParam: ptr.String("-updated"),
+			want: []string{
+				"Test 23",
+				"Test 22",
+				"Test 24",
+			},
+		},
+	}
+
+	svc, ctrl := rest.SecuredNamedSpaceController(identity)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// list all the spaces created by fixture helper using the
+			// code that we have written
+			_, spaces := test.ListNamedspacesOK(t, svc.Context, svc, ctrl,
```

Comment:
Maybe it makes sense to create a golden file for each search here. The response does contain a "link" to itself together with the sort parameter, right? This would be a very valuable resource for any client to look at and find out what parameters are possible. Also we would find out if we break the API at some point.

I know that you have defined the sort parameter to be an enum but still we will not recognize a breaking API change just with that.